open_lm/file_utils.py

```diff
@@ -181,7 +192,108 @@ def source_exhausted(paths, shard_list_per_source):
     return False
 
 
-def get_string_for_epoch(num_samples, starting_chunk, paths, weights, min_shards_needed):
+def count_small_shards(path, ratio=0.9):
+    shard_sizes = []
```

Comment:
"""Count the number of shards with significantly fewer sequences than the largest shard."""